In [2]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

# Adjust the path if your data is in the parent directory
df_long = pd.read_csv('/Users/sissili/Desktop/QTM350/QTM_350_Final_Project/QTM350_Final_Project_Data/cleaned_final_project_data_long.csv') 

# Display the first few rows to confirm column names
df_long.head()

,Country,year,gdp,primary_edu,secondary_edu,fertility
0,Brazil,YR2020,7074.194075,105.912231,103.882660,47.460
1,Brazil,YR2021,7972.536961,103.929398,106.298462,45.670
2,Brazil,YR2022,9281.332821,104.667419,105.702370,42.881
3,Brazil,YR2023,10377.589772,NaN,NaN,42.686
4,Brazil,YR2024,10310.548878,104.296913,104.144661,41.459


In [7]:
import statsmodels.api as sm
import numpy as np

# 1. Create Log GDP
# We use the log of GDP because wealth varies so much between Switzerland and Niger
df_long['log_gdp'] = np.log(df_long['gdp'])

# 2. Handle missing data
# Regression will fail if there are any NaN values in these specific columns
df_model = df_long.dropna(subset=['fertility', 'secondary_edu', 'log_gdp']).copy()

# 3. Define the Independent Variables (X) and Target (y)
X = df_model[['secondary_edu', 'log_gdp']]
y = df_model['fertility']

# 4. Add the constant (intercept)
X = sm.add_constant(X)

# 5. Fit the model
model = sm.OLS(y, X).fit()

# 6. Print the summary results
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:              fertility   R-squared:                       0.571
Model:                            OLS   Adj. R-squared:                  0.532
Method:                 Least Squares   F-statistic:                     14.62
Date:                Mon, 20 Apr 2026   Prob (F-statistic):           9.13e-05
Time:                        12:43:51   Log-Likelihood:                -116.00
No. Observations:                  25   AIC:                             238.0
Df Residuals:                      22   BIC:                             241.7
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const           160.2506     34.407      4.657

### Interpretation of the model:

Looking at the R square 0.571, which showing moderate power of the model in explanaing the variation in fertility.

F-statistic has a very significant p-value, which suggests that your model is much better than just guessing the average fertility rate.

Secondary Education (p=0.015): This value is below 0.05, making it statistically significant. The coefficient of −0.9465 means that for every 1-unit increase in secondary enrollment, the adolescent fertility rate is predicted to drop by almost 1 full birth per 1,000 women. 

Log GDP (p=0.293): This value is much higher than 0.05. This tells us that, once you account for education levels, wealth (GDP) does not have a statistically significant independent effect on fertility in this specific sample.

Therefore, our findings suggest that education is a more powerful tool for reducing adolescent fertility than economic growth alone. While wealthier nations tend to have lower fertility, the regression shows that it is the educational attainment of the population, rather than the raw GDP, that drives the change. In other words, for the countries studied, the 'protective effect' of keeping girls in school is a more direct and reliable indicator of lower fertility rates than national economic wealth. 